userguide/tutorials/invoice_plugin.adoc

```diff
@@ -37,7 +37,8 @@ When returning `InvoiceItem` objects back to Kill Bill, it is recommended to set
 
 However:
 
-* For new invoices or if an invoice is still in `DRAFT` mode, invoice plugins are allowed to update existing items, instead of adding new ones, by respecting the invoice item id returned by Kill Bill. This can be useful for instance to update tax amounts.
+* For new invoices or if an invoice is still in `DRAFT` mode, invoice plugins are allowed to update existing items, instead of adding new ones, by respecting the invoice item id returned by Kill Bill. This can be useful for instance to update tax amounts. 
+* It is worth noting that, specifying fields like `linkedInvoiceItemId`, `subscriptionId` will have no effect while adding invoice items of the type `EXTERNAL_CHARGE`, `CREDIT_ADJ`, `TAX`. These fields do not get saved on the database and will not be returned in subsequent `GET` requests. 
```

Comment:
@reshmabidikar Actually, I just checked: `linkedInvoiceItemId` can be populated by an invoice plugin (the Avatax plugin does it for instance).

Would be worth double checking the above (I suspect the behavior is different than the REST API).